rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java

```diff
@@ -199,6 +202,19 @@ public void maybeUpdateModel() {
         doAfterVisit(new UpdateMavenModel<>());
     }
 
+    public boolean isAnyPluginTag(
+        String groupId, String artifactId) {
+        Cursor cursor = getCursor();
+        for (XPathMatcher matcher : Arrays.asList(PLUGIN_MATCHER, MANAGED_PLUGIN_MATCHER,
+            PROFILE_PLUGIN_MATCHER, PROFILE_MANAGED_PLUGIN_MATCHER)) {
+            if (matcher.matches(cursor) && hasPluginGroupId(groupId) && hasPluginArtifactId(
+                artifactId)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
nit-pick: you can convert these to
```
```suggestion
        return Arrays.asList(PLUGIN_MATCHER, MANAGED_PLUGIN_MATCHER,
            PROFILE_PLUGIN_MATCHER, PROFILE_MANAGED_PLUGIN_MATCHER)
            .stream()
            .anyMatch(matcher -> matcher.matches(cursor) && hasPluginGroupId(groupId) && hasPluginArtifactId(
                artifactId));
```

Reply:
While that does help the readability, we try to limit our use of the streams API, especially in code paths that might get evaluated often, as the object allocations can hurt performance when running at scale.

Reply:
makes total sense, didn’t think about the performance of it